src/hooks/useReplyBox.tsx

```diff
@@ -0,0 +1,284 @@
+  return `tickets/${ticketId}/${APP_PREFIX}/notes/selection/${taskId}`.toLowerCase();
+};
+
+const emailKey = (ticketId: string, taskId: ITaskFromList["id"]|"*") => {
+  return `tickets/${ticketId}/${APP_PREFIX}/emails/selection/${taskId}`.toLowerCase();
+};
+
+const registerReplyBoxNotesAdditionsTargetAction = (
+  client: IDeskproClient,
+  ticketId: TicketData["ticket"]["id"],
+  taskIds: Array<ITaskFromList["id"]>,
+  tasksMap: Record<ITaskFromList["id"], ITaskFromList>,
+): void|Promise<void> => {
+  if (!ticketId) {
+    return;
+  }
+
+  if (Array.isArray(taskIds) && !taskIds.length) {
+    return client.deregisterTargetAction(`${APP_PREFIX}ReplyBoxNoteAdditions`);
+  }
+
+  return Promise
+    .all(taskIds.map((taskId: ITaskFromList["id"]) => client.getState<{ selected: boolean }>(noteKey(ticketId, taskId))))
+    .then((flags) => {
+      client.registerTargetAction(`${APP_PREFIX}ReplyBoxNoteAdditions`, "reply_box_note_item_selection", {
+        title: "Add to Wrike",
+        payload: taskIds.map((taskId, idx) => {
+          const task = tasksMap[taskId];
+          return {
+            id: task.id,
+            title: truncate(task.title),
+            selected: flags[idx][0]?.data?.selected ?? false,
+          };
+        }),
+      });
+    })
+    ;
+};
+
+const registerReplyBoxEmailsAdditionsTargetAction = (
+  client: IDeskproClient,
+  ticketId: TicketData["ticket"]["id"],
+  taskIds: Array<ITaskFromList["id"]>,
+  tasksMap: Record<ITaskFromList["id"], ITaskFromList>,
+): void|Promise<void> => {
+  if (!ticketId) {
+    return;
+  }
+
+  if (Array.isArray(taskIds) && !taskIds.length) {
+    return client.deregisterTargetAction(`${APP_PREFIX}ReplyBoxEmailAdditions`);
+  }
+
+  return Promise
+    .all(taskIds.map((taskId: ITaskFromList["id"]) => {
+      return client.getState<{ selected: boolean }>(emailKey(ticketId, taskId))
+    }))
+    .then((flags) => {
+      return client.registerTargetAction(`${APP_PREFIX}ReplyBoxEmailAdditions`, "reply_box_email_item_selection", {
+        title: `Add to Wrike`,
+        payload: taskIds.map((taskId, idx) => {
+          const task = tasksMap[taskId];
+          return {
+            id: task.id,
+            title: truncate(task.title),
+            selected: flags[idx][0]?.data?.selected ?? false,
+          };
+        }),
+      });
+    });
+};
+
+const ReplyBoxContext = createContext<ReturnUseReplyBox>({
+  setSelectionState: () => {},
+  getSelectionState: () => {},
+  deleteSelectionState: () => {},
+});
+
+const useReplyBox = () => useContext<ReturnUseReplyBox>(ReplyBoxContext);
+
+const ReplyBoxProvider: FC<PropsWithChildren> = ({ children }) => {
+  const { context } = useDeskproLatestAppContext();
+  const { client } = useDeskproAppClient();
+  const { tasks } = useLinkedTasks();
+  const tasksMap = useMemo(() => tasks.reduce<Record<ITaskFromList["id"], ITaskFromList>>((acc, task) => {
+      acc[task.id] = task;
+      return acc;
+    }, {}), [tasks]);
```

Comment:
No need for a memo here, creating a map out of an array of tasks is cheap for the CPU to do - even if large.

Reply:
fixed